flink-formats/flink-orc/src/main/java/org/apache/flink/orc/OrcFileSystemFormatFactory.java

```diff
@@ -83,14 +96,227 @@ private static Properties getOrcProperties(ReadableConfig options) {
 		return orcProperties;
 	}
 
+	private boolean isUnaryValid(CallExpression callExpression) {
+		return callExpression.getChildren().size() == 1 && callExpression.getChildren().get(0) instanceof FieldReferenceExpression;
+	}
+
+	private boolean isBinaryValid(CallExpression callExpression) {
+		return callExpression.getChildren().size() == 2 && ((callExpression.getChildren().get(0) instanceof FieldReferenceExpression && callExpression.getChildren().get(1) instanceof ValueLiteralExpression) ||
+				(callExpression.getChildren().get(0) instanceof ValueLiteralExpression && callExpression.getChildren().get(1) instanceof FieldReferenceExpression));
+	}
+
+	public OrcSplitReader.Predicate toOrcPredicate(Expression expression) {
+		if (expression instanceof CallExpression) {
+			CallExpression callExp = (CallExpression) expression;
+			FunctionDefinition funcDef = callExp.getFunctionDefinition();
+
+			if (funcDef == BuiltInFunctionDefinitions.IS_NULL || funcDef == BuiltInFunctionDefinitions.IS_NOT_NULL || funcDef == BuiltInFunctionDefinitions.NOT) {
```

Comment:
Maybe we can have a `static final ImmutableMap<FunctionDefinition, Function<CallExpression, Expression>> FILTERS`. The function style can make codes better.

Reply:
do you means: static final ImmutableMap<FunctionDefinition, Function<CallExpression, OrcSplitReader.Predicate>> FILTERS ?

Reply:
Yes